File: src/Modules/Contracts/Core.Contract/DTOs/Operator/OperatorDto.cs
using SharedKernel.Enums;

namespace Core.Contract.DTOs.Operator;

public sealed record OperatorDto(
      Guid Guid,
  string Username = "",
  Title Title = Title.Mr,
  string FirstName = "",
  string MiddleName = "",
  string LastName = "",
  Gender Gender = Gender.Male,
  string Email = "",
  string Phone = "",
  Guid RoleGuid = default!,
  string Role = "",
  DateTime JoinedDate = default!,
  DateTime ExpiredDate = default!,
  List<Guid> LocationGuids = default!,
  bool IsActive = false,
  bool IsDefault = false
);

File: src/Web/src/model/Operator/OperatorDto.ts
import { Gender } from "../../enum/Gender";
import { Title } from "../../enum/Title";


export interface OperatorDto  {
    guid:string;
    username:string;
    password:string;
    title:Title;
    firstname:string;
    middlename:string;
    lastname:string;
    gender:Gender;
    email:string;
    mobile:string;
    roleGuid:string; 
    locationGuids:string[]; 
    isActive:boolean;
    isDefault:boolean;
}
